packages/paymaster/src/BiconomyPaymaster.ts

```diff
@@ -25,20 +25,24 @@ export class BiconomyPaymasterAPI implements IPaymasterAPI {
       userOp.paymasterAndData = '0x'
 
       const result: PaymasterAndDataResponse = await sendRequest({
-        url: this.paymasterServiceUrl,
+        url: this.paymasterConfig.paymasterServiceUrl,
         method: HttpMethod.Post,
+        headers: { 'x-api-key': this.paymasterConfig.dappAPIKey },
         body: { userOp: userOp }
       })
 
       if (result && result.data && result.statusCode === 200) {
         return result.data.paymasterAndData
       } else {
-        if (!this.strictSponsorshipMode) {
+        if (!this.paymasterConfig.strictSponsorshipMode) {
           return '0x'
         }
       }
     } catch (error) {
-      console.log('error while getting paymaster response')
+      console.log('error while getting paymaster response', error)
```

Comment:
Change to console.error and start the line with capital letter, "Error while .... " instead of "error while ...."
And move it inside next if statement.